src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/AbstractRazorDelegatingEndpoint.cs

```diff
@@ -166,7 +167,8 @@ protected virtual Task<TResponse> HandleDelegatedResponseAsync(TResponse delegat
         }
         catch (RemoteInvocationException e)
         {
-            requestContext.Logger.LogException(e);
+            Logger.LogError(e, "Error calling delegate server for {method}", CustomMessageTarget);
+            requestContext.GetRequiredService<ITelemetryReporter>().ReportFault(e, "Error calling delegate server for {method}", CustomMessageTarget);
```

Comment:
The one thing we lose when moving away from `IRazorLogger`/`ILspLogger` here is the ability to log exceptions. There were only two places that did it though, so I just made them explicitly report telemetry then log the error, which is all `LoggerAdapter` did anyway.

In future it would be good to fix up this too, so that we had a single logger provider that just reported telemetry, but that doesn't work without more cleanup, because right now we only report telemetry when `LogError` is called from CLaSP. Separating though out with generic `ILogger` needs though (eg. do we always report telemetry for LogError calls, and just audit all of the existing ones we have?)

Reply:
Should we log an error to remember this for later?

Reply:
This still does log an error, on line 170, so there is no changed to behaviour just slightly less convenient code

Reply:
ITelemetryReporter could probably also call `LogError` on faults

Reply:
OMG talk about not being able to see the forest for the trees 🤦‍♂️ 

Thank you, thats a very nice solution. Will do in a follow up though :)

Reply:
Actually, this might not be so good, because the log message would say `[TelemetryReporter]` at the start, rather than with the current code, which at least says the name of the endpoint.

There were only two points where `LogException` was called though, so perhaps we just leave this for a bit.